test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightDeltaReaderTests.cs

```diff
@@ -877,18 +923,21 @@ public void ReadNestedDeletedEntryIn41DeltaResource()
             nestedResourceInfo.Name.Should().Be("ProductBeingViewed");
             Assert.NotNull(nestedDeletedResource);
             nestedDeletedResource.Reason.Should().Be(DeltaDeletedEntryReason.Deleted);
-            nestedDeletedResource.Id.Should().Be(new Uri("http://host/service/Products/10"));

```

Comment:
>Id. [](start = 34, length = 3)

The "Id" will never be set?

Reply:
Yes; previously we were generating the Id based on convention, but I realized that logic was flawed. In a 4.01 delta payload, the id can be omitted if the key values are provided. In a normal response payload that implies that the id is generated by convention, but in a delta payload omitting the id doesn't say anything about whether the syntax follows convention or not. So materializing the id based on the key values was presuming the service followed convention, which was not necessarily the case and could have resulted in invalid ids.  So Id will now be set in a 4.01 delta payload only if @odata.id is explicitly specified in the payload (which it is not in these tests).

---
In reply to: [256643472](https://github.com/OData/odata.net/pull/1404#discussion_r256643472) [](ancestors = 256643472)